### PR TITLE
Test fails on evals/requires

### PR DESCRIPTION
**Pull request by flbulgarelli**
Although we should simply abort evaluation of submissions with potential malicious  code instead of making test fail, the proposed approach in this PR is better than nothing.

### CODE REVIEW

**Comment by flbulgarelli** [comment]
@faloi Please comment

File: src/server/test-compiler.es6 [hunk 1]
```diff
@@ -23,6 +23,10 @@ function writeContent(path, content) {
 function compile(body) {
   return `
     var assert = require("assert");
+    require = function() { throw new Error("require not available") };
+    module.require = require;
+    eval = function() { throw new Error("eval not available") };
```
Comment (faloi) [inline src/server/test-compiler.es6:28]: This one should be tested too

File: src/server/test-compiler.es6 [hunk 2]
```diff
@@ -23,6 +23,10 @@ function writeContent(path, content) {
 function compile(body) {
   return `
     var assert = require("assert");
+    require = function() { throw new Error("require not available") };
+    module.require = require;
```
Comment (faloi) [inline src/server/test-compiler.es6:27]: Why is this necessary?

Comment (flbulgarelli) [inline src/server/test-compiler.es6:27]: There is module.require method which is somewhat a backdoor:

https://nodejs.org/api/modules.html#modules_module_require_id

I am not completly sure this could be a problem, but just in case...
